ClimsoftVer4/ClimsoftVer4/ucrNavigation.vb

```diff
@@ -318,4 +332,20 @@
 
     End Sub
 
+    ' Use these two methods when you need to get a values from a specific row of the table
+    ' These should be used in any place where dtbRecords is currently used since we are now not populating dtbRecords
+    Private Function GetValueFromRow(iRow As Integer, strField As String) As String
+        If iMaxRows = 0 Then
+            Return ""
+        Else
+            Return CallByName(GetRow(iRow), strField, CallType.Get)
+        End If
+    End Function
+
+    Private Function GetRow(iRow As Integer) As Object
+        'Skip() and FirstOrDefault() seems like the way to get the nth row from the table
+        'For some reason you can only use Skip() if you use an Order function first.
+        'We might want to sort the records for the sequencer anyway?
+        Return clsDataConnection.db.form_daily2.OrderByDescending(Function(u) u.stationId).Skip(iRow).FirstOrDefault()
+    End Function
```

Comment:
This of course also needs to be done in general as above probably using the trycast

Reply:
@dannyparsons this minor change will solve the issue of EF tracking.

```
Private Function GetRow(iRow As Integer) As Object
        'Skip() and FirstOrDefault() seems like the way to get the nth row from the table
        'You can only use Skip() if you use an Order function first.
        Dim x = CallByName(clsDataConnection.db, clsDataDefinition.GetTableName(), CallType.Get)
        x = x.AsNoTracking()
        Dim y = TryCast(x, IQueryable(Of Object))
        ' OrderBy function returns 1 to give a default ordering
        Return y.OrderBy(Function(u) 1).Skip(iRow).FirstOrDefault()
  End Function
```